packages/renderer/src/lib/image/PullImage.spec.ts

```diff
@@ -475,3 +484,158 @@ describe('container connections', () => {
     });
   });
 });
+
+describe('Preferred Registries', () => {
+  beforeEach(() => {
+    vi.mocked(window.getConfigurationValue).mockImplementation(async (key: string) => {
+      if (key === 'terminal.integrated.scrollback') {
+        return 1000;
+      }
+      if (key === 'registries.preferredRegistries') {
+        return 'quay.io, ghcr.io, docker.io';
+      }
+      return undefined;
+    });
+  });
+
+  test('should load preferred registries from configuration on mount', async () => {
+    render(PullImage);
+
+    await vi.waitFor(() => {
+      expect(window.getConfigurationValue).toHaveBeenCalledWith('registries.preferredRegistries');
+    });
+  });
+
+  test('should search all preferred registries when no registry prefix specified', async () => {
+    vi.mocked(window.searchImageInRegistry).mockImplementation(async options => {
+      if (options.registry === 'quay.io') {
+        return [
+          { name: 'nginx', description: '', star_count: 0, is_official: false },
+          { name: 'redis', description: '', star_count: 0, is_official: false },
+        ];
+      }
+      if (options.registry === 'ghcr.io') {
+        return [
+          { name: 'nginx', description: '', star_count: 0, is_official: false },
+          { name: 'postgres', description: '', star_count: 0, is_official: false },
+        ];
+      }
+      if (options.registry === 'docker.io') {
+        return [
+          { name: 'nginx', description: '', star_count: 0, is_official: false },
+          { name: 'mysql', description: '', star_count: 0, is_official: false },
+        ];
+      }
+      return [];
+    });
+
+    render(PullImage);
+    await tick();
```

Comment:
question: why do you need to use tick()?

If we need to wait for something (E.g. the textbox) to be displayed, we should prefer `vi.waitFor`